pkg/config/templateComponent.go

```diff
@@ -53,6 +53,15 @@ type TemplateData struct {
 	Data   []any
 }
 
+// ComponentValidation represents a validation rule that applies to multiple properties
+type ComponentValidation struct {
```

Comment:
This is not consistent with the way it's handled at the property level. Those take an optional list of comma-separated arguments, and we simply construct the error message by naming the validation that failed. I think we should do the same here, for usage consistency when writing components, as well as client-side code consistency (we depend on that for constructing the user-visible error message).